app/services/task_list_creator.rb

```diff
@@ -11,19 +11,25 @@ def call(project)
   end
 
   private def create_task_list_from_workflow(project, workflow)
-    workflow.fetch("sections").each_with_index do |workflow_section, index|
```

Comment:
You seem to have changed more than the optional values here? Raising a `KeyNotFound` error might be more helpful than returning `nil` and having unexpected behaviour elsewhere?

If you want to use fetch with a default you can:

`fetch("sections", nil)`

Reply:
Updated them all back to `fetch`, providing a default value of `nil` when the field isn't declared.

I've also added a new mock workflow without any of the optional values so we can test it explicitly. I could have used the existing mock workflow, but this seems more obvious to me 🤔